cpp/tests/join/point_in_polygon_test_large.cpp

```diff
@@ -235,57 +172,48 @@ TYPED_TEST(PIPRefineTestLarge, TestLarge)
-                                       cudf::test::to_host<T>(poly_y).first);
-
-  auto host_poly_and_point_indices =
-    geometry_to_poly_and_point_indices(h_poly,
-                                       cudf::test::to_host<T>(points->get_column(0)).first,
-                                       cudf::test::to_host<T>(points->get_column(1)).first);
-
-  auto& expected_poly_indices  = std::get<0>(host_poly_and_point_indices);
-  auto& expected_point_indices = std::get<1>(host_poly_and_point_indices);
-  auto& expected_point_lengths = std::get<2>(host_poly_and_point_indices);
-
   auto actual_poly_indices  = cudf::test::to_host<uint32_t>(poly_idx).first;
   auto actual_point_indices = cudf::test::to_host<uint32_t>(point_idx).first;
-  auto actual_point_lengths = thrust::host_vector<uint32_t>(point_in_polygon_pairs->num_rows());
 
   thrust::stable_sort_by_key(
     actual_point_indices.begin(), actual_point_indices.end(), actual_poly_indices.begin());
 
-  auto num_search_points = thrust::distance(actual_point_indices.begin(),
-                                            thrust::reduce_by_key(actual_point_indices.begin(),
-                                                                  actual_point_indices.end(),
-                                                                  thrust::make_constant_iterator(1),
-                                                                  actual_point_indices.begin(),
-                                                                  actual_point_lengths.begin())
-                                              .first);
+  {
+    // verify
+
+    auto hits = cuspatial::point_in_polygon(
+      points->get_column(0), points->get_column(1), poly_offsets, ring_offsets, poly_x, poly_y);
 
-  actual_point_indices.resize(num_search_points);
-  actual_point_lengths.resize(num_search_points);
-  actual_point_indices.shrink_to_fit();
-  actual_point_lengths.shrink_to_fit();
+    auto hits_host = cudf::test::to_host<int32_t>(hits->view()).first;
 
-  auto poly_a = fixed_width_column_wrapper<uint32_t>(expected_poly_indices.begin(),
-                                                     expected_poly_indices.end());
-  auto poly_b =
-    fixed_width_column_wrapper<uint32_t>(actual_poly_indices.begin(), actual_poly_indices.end());
-  CUDF_TEST_EXPECT_COLUMNS_EQUAL(poly_a, poly_b, verbosity);
+    std::vector<uint32_t> expected_poly_indices;
+    std::vector<uint32_t> expected_point_indices;
 
-  auto point_a = fixed_width_column_wrapper<uint32_t>(expected_point_indices.begin(),
-                                                      expected_point_indices.end());
-  auto point_b =
-    fixed_width_column_wrapper<uint32_t>(actual_point_indices.begin(), actual_point_indices.end());
-  CUDF_TEST_EXPECT_COLUMNS_EQUAL(point_a, point_b, verbosity);
+    for (int point_index = 0; point_index < hits->size(); point_index++) {
+      // iterate over set bits
+      std::uint32_t bits = hits_host[point_index];
```

Comment:
Small nit: `bits` is unsigned here, but `hits_host` is signed.

Maybe a larger question, but why does `cuspatial::point_in_polygon` return an INT32 column vs. a UINT32? Does the sign bit indicate something special? If not, unsigned conveys the sign bit isn't special and the column is a bitmask.

Alternatively, we could return a column with empty `data` and a full `null_mask`. This would let us test whether values were null without reading the bits ourselves?

Reply:
OMG, I think it may be because cuDF didn't support unsigned types until here: https://github.com/rapidsai/cudf/pull/5292